src/nvim/fold.c

```diff
@@ -1111,7 +1111,7 @@ static int foldLevelWin(win_T *wp, linenr_T lnum)
 /// Check if the folds in window "wp" are invalid and update them if needed.
 static void checkupdate(win_T *wp)
 {
-  if (!wp->w_foldinvalid) {
+  if (!wp->w_foldinvalid || wp->w_buffer->b_ml.ml_flags & ML_EMPTY) {
```

Comment:
As [before](https://github.com/neovim/neovim/pull/32233#issuecomment-2618393809), I'm still not sure _how_ this leaves our treesitter modules in a broken state. This prevents us from calling `foldexpr()` for a window into an empty buffer, which also happens to be the case while redrawing during the swapfile prompt.

Reply:
I've moved this check to `vim.treesitter.foldexpr()` because an empty buffer can (for some reason) still have folds which is tested by both diffmode and foldexpr tests.

I still suspect this case should actually be handled better by `languagetree.lua` but I haven't been able to figure out how.